lib/src/entrypoint.dart

```diff
@@ -24,33 +23,30 @@ import 'package.dart';
 import 'package_name.dart';
 import 'package_graph.dart';
 import 'pubspec.dart';
-import 'sdk.dart' as sdk;
+import 'sdk.dart';
 import 'solver.dart';
 import 'source/cached.dart';
 import 'source/unknown.dart';
 import 'system_cache.dart';
 import 'utils.dart';
 
-/// A RegExp to match the Dart SDK constraint in a lock file.
-///
-/// This matches both the old-style constraint:
-///
-/// ```yaml
-/// sdk: ">=1.2.3 <2.0.0"
-/// ```
-///
-/// and the new-style constraint:
-///
-/// ```yaml
-/// sdks:
-///   dart: ">=1.2.3 <2.0.0"
-/// ```
-final _dartSdkConstraint =
-    new RegExp(r'^(  dart|sdk): "?([^"]*)"?$', multiLine: true);
-
-/// A RegExp to match the Flutter SDK constraint in a lock file.
-final _flutterSdkConstraint =
-    new RegExp(r'^  flutter: "?([^"]*)"?$', multiLine: true);
+/// A RegExp to match SDK constraints in a lockfile.
+final _sdkConstraint = () {
+  // This matches both the old-style constraint:
+  //
+  // ```yaml
+  // sdk: ">=1.2.3 <2.0.0"
+  // ```
+  //
+  // and the new-style constraint:
+  //
+  // ```yaml
+  // sdks:
+  //   dart: ">=1.2.3 <2.0.0"
+  // ```
+  var sdkNames = sdks.keys.map((name) => "  " + name).join('|');
+  return new RegExp(r'^(' + sdkNames + r'|sdk): "?([^"]*)"?$', multiLine: true);
```

Comment:
Perhaps a bit out of scope for the review - but why do we need this regex to be multiLine?